### PR TITLE
fix(web): use punycode for domains

### PR DESCRIPTION
**Pull request by antonymarion**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-12477
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)

## Description

Convert to Unicode in several places (for displayed domain name for instance)

### CODE REVIEW

File: packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
         ></a>
         <span
             data-ng-if="!(element.url && !element.active && !$last)"
-            data-ng-bind="element.name"
+            data-ng-bind="$ctrl.convertDomainToPunycode(element.name)"
```
Comment (qpavy) [inline packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html:10]: Does this work as it is a static class method ?
Maybe you should use constructor to access this method
```suggestion
            data-ng-bind="$ctrl.constructor.convertDomainToPunycode(element.name)"
```

File: packages/manager/apps/web/client/app/dns-zone/zone/dns-zone.html [hunk 2]
```diff
@@ -11,7 +11,7 @@
                 <h1
                     class="oui-header__title"
                     data-translate="domain_tab_ZONE_named"
-                    data-translate-values="{ t0: ctrlDomain.domain.displayName }"
+                    data-translate-values="{ t0: ctrlDomain.convertDomainToPunycode(ctrlDomain.domain.displayName) }"
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/dns-zone/zone/dns-zone.html:14]: same as https://github.com/ovh/manager/pull/10173/files#r1371319204

File: packages/manager/apps/web/client/app/domain/list/template.html [hunk 3]
```diff
@@ -33,7 +33,7 @@
     >
         <a
             data-ng-href="{{:: $ctrl.getServiceNameLink($row) }}"
-            data-ng-bind="::$value"
+            data-ng-bind="{{::$ctrl.convertDomainToPunycode($value)}}"
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/domain/list/template.html:36]: same as https://github.com/ovh/manager/pull/10173/files#r1371319204

File: packages/manager/modules/hub/src/components/products/products.html [hunk 4]
```diff
@@ -18,7 +18,7 @@
                 >
                     <a
                         data-ng-href="{{:: service.url }}"
-                        data-ng-bind="service.resource.displayName || service.resource.name"
+                        data-ng-bind="$ctrl.convertDomainToPunycode(service.resource.displayName || service.resource.name)"
```
Comment (qpavy) [inline packages/manager/modules/hub/src/components/products/products.html:21]: same as https://github.com/ovh/manager/pull/10173/files#r1371319204

File: packages/manager/modules/ng-layout-helpers/src/list/list-layout.utils.js [hunk 5]
```diff
@@ -36,7 +37,7 @@ export const mapFilterForIceberg = (comparator, reference) =>
 export const getLink = (column, tracker) => `
   <a
     data-ng-href="{{ $ctrl.getServiceNameLink($row) }}"
-    data-ng-bind="$row.${column.property}"
+    data-ng-bind="$ctrl.convertDomainToPunycode($row.${column.property})"
```
Comment (qpavy) [inline packages/manager/modules/ng-layout-helpers/src/list/list-layout.utils.js:40]: This column can be other things that a Domain. Maybe it is better to change the function name ?

File: packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html (hunk 1 above)
Comment (antonymarion) [reply packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html:10]: Yes it worked and was tested (cf Jira). Let me check another time :)

File: packages/manager/modules/ng-layout-helpers/src/list/list-layout.utils.js (hunk 5 above)
Comment (antonymarion) [reply packages/manager/modules/ng-layout-helpers/src/list/list-layout.utils.js:40]: But I wanted to keep it consistent with original namings (could be usefull in the future to have same function name with same intention everywhere in the code).

What do you think?

File: packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html (hunk 1 above)
Comment (antonymarion) [reply packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html:10]: Seems ok also in the deployed version:
image

Comment (antonymarion) [reply packages/components/ng-ui-router-breadcrumb/src/ng-ui-router-breadcrumb.html:10]: You're right, thks, let me correct this!

File: packages/manager/modules/ng-layout-helpers/src/list/list-layout.utils.js (hunk 5 above)
Comment (qpavy) [reply packages/manager/modules/ng-layout-helpers/src/list/list-layout.utils.js:40]: Maybe you can change it everywhere by "convertToPunycode ?

**Review by qpavy — APPROVED** [review]

**Pull request merged**